File: src/main/java/com/fengsheng/handler/end_receive_phase_tos.java
package com.fengsheng.handler;

import com.fengsheng.HumanPlayer;
import com.fengsheng.protos.Fengsheng;
import org.apache.log4j.Logger;

public class end_receive_phase_tos extends AbstractProtoHandler<Fengsheng.end_receive_phase_tos> {
    private static final Logger log = Logger.getLogger(end_receive_phase_tos.class);

    @Override
    protected void handle0(HumanPlayer r, Fengsheng.end_receive_phase_tos pb) {
        if (!r.checkSeq(pb.getSeq())) {
            log.error("操作太晚了, required Seq: " + r.getSeq() + ", actual Seq: " + pb.getSeq());
            return;
        }
        r.getGame().tryContinueResolveProtocol(r, pb);
    }
}

File: src/main/kotlin/handler/end_receive_phase_tos.kt
package com.fengsheng.handler

import com.fengsheng.HumanPlayer
import com.fengsheng.protos.Fengsheng
import org.apache.log4j.Logger

class end_receive_phase_tos : AbstractProtoHandler<Fengsheng.end_receive_phase_tos>() {
    override fun handle0(r: HumanPlayer, pb: Fengsheng.end_receive_phase_tos) {
        if (!r.checkSeq(pb.seq)) {
            log.error("操作太晚了, required Seq: ${r.seq}, actual Seq: ${pb.seq}")
            return
        }
        r.game!!.tryContinueResolveProtocol(r, pb)
    }

    companion object {
        private val log = Logger.getLogger(end_receive_phase_tos::class.java)
    }
}